paddle/operators/dynamic_recurrent_op.h

```diff
@@ -145,27 +165,49 @@ class DynamicRecurrentOp : public framework::OperatorBase {
   };
 
  private:
-  std::unique_ptr<OperatorBase> stepnet_;
-  mutable std::map<std::string, framework::TensorArray> states_;
-  mutable std::map<std::string, framework::TensorArray> step_inputs_;
-  mutable std::map<std::string, framework::TensorArray> step_outputs_;
-  mutable std::map<std::string, std::vector<framework::DySeqMeta>>
-      dy_seq_metas_;
-  mutable rnn::Argument arg_;
-  mutable ArgCache cache_;
+  std::unique_ptr<framework::OperatorBase> stepnet_;
+  std::map<std::string, framework::TensorArray> states_;
+  std::map<std::string, framework::TensorArray> step_inputs_;
+  std::map<std::string, framework::TensorArray> step_outputs_;
+  std::map<std::string, std::vector<framework::DySeqMeta>> dy_seq_metas_;
+  rnn::Argument arg_;
+  ArgCache cache_;
```

Comment:
What is the purpose of `ArgCache`? I saw that the terminology in your code includes "inlinks", "outlink", and "memory". Are they in "ArgCache", or what is the relationship between them and "argument"?

Reply:
This is a cache of all the arguments that located in the parent scope. RNN's computation is split into a bunch of protected methods, it is painful and ugly to pass in the arguments throughout all the member functions, so a cache called `ArgCache` is added as a member.

There is only one public member function of RNNAlgorthm, which is `Run`, refresh the `ArgCache` when `Run` is invoked.